### ISSUE TITLE
Output.file creating 0 byte files during network outages

### ISSUE THREAD

**Issue by MediaMeasure** [savonet/liquidsoap#788]
**Describe the bug**
My Liquidsoap script is setup on a high level as follows:
```
radio_stream = input.alsa
radio_stream = strip_blank

output.icecast(
  radio_stream,
  fallible=true,
  etc..
)
output.file(%mp3, radio_stream, fallible=true)
```

My issue is when strip blank brings up the input source during a network outage (im using using 3G...) both icecast and output.file do not work. output.file produces 0 byte files. 
 
**To Reproduce**
Create an input from alsa and then link it to strip blank, i.e. only stream on audio availability. Then create two outputs one icecast and one file. Disconnect network cable( to simulate network issue) and play some audio to bring up sorce, you will find that the file output is not working (making 0 byte files) as it seems to be linked to the connecting to icecast failure.

**Expected behavior**
Icecast should not stream as there is no network, but the file dump should be full for the duration that audio source is available as per strip_blank

**Version details**
 - OS: Ubuntu 16
 - Version 1.3.3

**Install method**
apt-get

**Comment by smimram** [comment]
I guess this is a case where clocks should be used (see [at the end of this documentation page](http://localhost:8000/doc-dev/clocks.html)).

**Comment by MediaMeasure** [comment]
Hi smimran,

Clocks looks like what I need however when trying to implement based on the document I get the following error:

Error when initializing source at line 28 char 16 - line 40 char 52:
  a source cannot belong to two clocks (wallclock_alsa[],
  wallclock_icecast[]).

Please advise.
Thanks

**Comment by smimram** [comment]
I can confirm that
```
s = single("test.mp3")
clock.assign_new([out(mksafe(buffer(s)))])
output.file(%mp3,"/tmp/xxx.mp3",s)
```
is not working whereas it should... I'll have a look.

**Comment by toots** [comment]
I think @dbaelde is out of commission for now. I'll look at it in the beta cycle.

**Comment by smimram** [comment]
I actually don't think that clocks are the right solution here, see the explanation in #288.

**Comment by dbaelde** [comment]
I could not reproduce the problem locally (by stopping my icecast server) and could only reproduce it non-deterministically with a remote icecast (by shutting down my wifi connection). The issue is perhaps due to `Cry.send` blocking when the connection is lost at the wrong time. In other cases the lost connection is cleanly detected and the icecast outputs stops while the file output keeps running as expected.

This may be considered as a bug, but it seems hard to fight against this, and in the end clocks are probably the way to go for a safe solution. I'll try to understand later why you ran into problems using them and propose a solution. (Regarding Samuel's code,  there is no bug as explained in #288.)

**Comment by toots** [comment]
I think this was closed with 6b9ed7e0a4fb932c5cce30e26e188556c7977f29
